EpsilonDriverControls/Src/DriverControls.c

```diff
@@ -52,6 +52,19 @@ float calculateMotorCurrent(float accelPercentage)
     }
 }
 
+float calculateAccelMotorCurrent(float accelPercentage, float prevMotorCurrent)
+{
+    // Smoothing output current so that there aren't any big jumps that could cause the motors to trip. Essentially a simple IIR low pass filter
+    return prevMotorCurrent + MOTOR_CURRENT_SMOOTHING_FACTOR * (calculateMotorCurrent(accelPercentage) - prevMotorCurrent);
+}
+
+float calculateRegenMotorCurrent(float regenPercentage, float prevMotorCurrent)
+{
+    // Smoothing output current so that there aren't any big jumps that could cause the motors to trip. Essentially a simple IIR low pass filter.
+    // Scaled by REGEN_INPUT_SCALING because regen uses less current than normal acceleration. Motors will trip if current is greater.
+    return prevMotorCurrent + MOTOR_CURRENT_SMOOTHING_FACTOR * (calculateMotorCurrent(regenPercentage) * REGEN_INPUT_SCALING - prevMotorCurrent);
+}
```

Comment:
I think we can extract the smoothing function out again:
```
float smoothCurrent(float prevMotorCurrent, float presentMotorCurrent)
{
    // Smoothing output current so that there aren't any big jumps that could cause the motors to trip. Essentially a simple IIR low pass filter
     return prevMotorCurrent + MOTOR_CURRENT_SMOOTHING_FACTOR * (presentMotorCurrent- prevMotorCurrent);
}
float calculateAccelMotorCurrent(float accelPercentage, float prevMotorCurrent)
{
    return smoothCurrent(prevMotorCurrent, calculateMotorCurrent(accelPercentage));
}

float calculateRegenMotorCurrent(float regenPercentage, float prevMotorCurrent)
{
    float presentRegenCurrent = calculateMotorCurrent(regenPercentage) * REGEN_INPUT_SCALING;
    return smoothCurrent(prevMotorCurrent, presentRegenCurrent);
}
```